### PR TITLE
[libc][bazel] Allow configure options to alter all targets

### PR DESCRIPTION
**Pull request by gchatelet**
The previous state was leading to inconsistencies. Some targets would get the options and some wouldn't. As an example, the `MEMORY_COPTS` definitions would only apply to the `:string_memory_utils` target but not to the `:memcpy` target. This patch makes sure definitions are applied throughout the LLVM libc targets as `local_defines`. This ensures that the preprocessor definitions don't propagate to depending targets outside of LLVM libc, and that all libc targets have consistent preprocessor definitions.

### CODE REVIEW

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel [hunk 1]
```diff
@@ -3228,7 +3213,7 @@ libc_support_library(
 libc_support_library(
     name = "printf_mock_parser",
     hdrs = ["src/stdio/printf_core/parser.h"],
-    defines = PRINTF_COPTS + ["LIBC_COPT_MOCK_ARG_LIST"],
+    defines = ["LIBC_COPT_MOCK_ARG_LIST"],
```
Comment (gchatelet) [inline utils/bazel/llvm-project-overlay/libc/BUILD.bazel:3216]: Note to libc maintainers: we should probably not use `defines` at all in our `BUILD` files as they will propagate up outside of the libc project /me think. Upon changes this potentially triggers a rebuild of all targets depending on libc. @michaelrj-google @nickdesaulniers @lntue

Comment (gchatelet) [reply utils/bazel/llvm-project-overlay/libc/BUILD.bazel]: Done in  https://github.com/llvm/llvm-project/pull/89251/commits/936169a3b93a2bbcd3a09057471244de2253649f

**Review by legrosbuffle — APPROVED** [review]

**Comment by gchatelet** [comment]
@michaelrj-google can you have a look as well?

**Review by michaelrj-google — APPROVED** [review]
moving away from `defines` to `local_defines` seems like a good idea. This patch LGTM

**Pull request merged**